src/main/java/edu/hm/hafner/analysis/parser/PyLintParser.java

```diff
@@ -19,9 +19,10 @@
     private static final long serialVersionUID = 4464053085862883240L;
 
     // the default pattern matches "--output-format=parseable" output.
-    private static final String PYLINT_PATTERN = "(?<path>[^:]*)(?:\\:(?<module>.*))?:(?<line>\\d+): \\[(?<category>\\D\\d*)(?:\\((?<symbol>.*)\\), )?.*?\\] (?<message>.*)";
+    private static final String PYLINT_PATTERN = "(?<path>[^:]*)(?:\\:(?<module>.*))?:(?<line>\\d+): \\[(?<type>(?<category>[A-Z])\\d*)?(?:\\((?<symbol>.*)\\), )?.*?\\] (?<message>.*)";
```

Comment:
The type (and so the category) group is now optional. That way, in the case where someone would not add `{msg_id}` in the `--msg-template`, we won't have a `java.util.NoSuchElementException` error.